### ISSUE TITLE
Add CLI

### ISSUE THREAD

**Issue by chrahunt** [chrahunt/quicken#21]
There's no reason we can't provide a CLI to speed up existing applications for end users. For example

```
# file path
$ quicken ./example.py
# modules
$ quicken -m pytest
```

For argument parsing, proceed as if all arguments after the path or `-m arg` are meant for the underlying command, for example:

```
$ quicken ./example.py arg1
# ->
$ ./example.py arg1
$ quicken -m pytest -s -ra
# ->
$ pytest -s -ra
```

For scripts:

1. Input is normalized to `path` then resolved to `real_path`
1. `name` is `quicken.cli.file.{digest}`, where `digest = sha256(path)`
1. `metadata`:
   1. `path`
   1. `real_path` (reload criteria)
   1. ctime of `real_path` (reload criteria)
   1. mtime of `real_path` (reload criteria)
1. function:
   1. parse the file at `real_path`, extract into prelude and `if __name__ == '__main__'` section. Execute prelude as preparation and return conditional section as the main function.

For modules:

1. resolve module
1. `name` is `quicken.cli.module.{digest}`, where `digest = sha256(module_path)`
1. `metadata`:
   1. `module_path`
   1. `module`
   1. last modified time of `module_path` (reload criteria)
1. function:
   1. parse `module`, getting imports manually
   1. return function that executes the rest of the module body of `__main__.py`

**Comment by chrahunt** [comment]
Still pending:

- [ ] module integration (`-m mod`)
- [ ] CLI troubleshooting (what not to do, e.g. imports of nested modules)

Possible implementation (also described in `quicken/_cli.py`):

1. generate digest
   1. using importlib, find module (currently nested modules are not supported)
1. generate prelude and main
   1. prelude
      1. extract all top-level import statements
      1. extract all import statements that are in an `if __name__ == '__main__'` block
      1. execute all top-level import statements
         1. as a guard against import order issues, we should check that the identity of all properties on all imported modules is the same after each import
   1. main
      1. assuming there are no import issues, as mentioned above, then we can probably just eval the module and don't need to split it up

**Comment by chrahunt** [comment]
Closed by #54.
